Source/Controllers/MainViewControllers/ConnectionView/SPConnectionController.m

```diff
@@ -2099,9 +2099,8 @@ - (void)initiateMySQLConnectionInBackground
 			}
 		}
 
-		// Only set the password if there is no Keychain item set or the connection is being tested.
-		// The connection will otherwise ask the delegate for passwords in the Keychain.
-		if ((!connectionKeychainItemName || isTestingConnection) && [self password]) {
+		// Only set the password if there is no Keychain item set or the connection is being tested or the password is different than in Keychain.
+		if ((isTestingConnection || !connectionKeychainItemName || (connectionKeychainItemName && ![[self password] isEqualToString:@"SequelAceSecretPassword"])) && [self password]) {
 			[mySQLConnection setPassword:[self password]];
```

Comment:
Should we perhaps do the same logic here for the mysql password that you did for the SSH password?

Reply:
I did it vice versa, I applied this logic to SSH tunnel :)